### PR TITLE
Give an option to also implement IEquatable when generating Equals/GetHashCode.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/17650

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.GenerateEqualsAndGetHashCodeFromMembers
 {
+    using static GenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider;
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: Using inside namespace - please just stahp!

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs [hunk 2]
```diff
@@ -1119,7 +1121,109 @@ public override bool Equals(object obj)
     }
 }",
 chosenSymbols: null,
-optionsCallback: options => options[0].Value = true,
+optionsCallback: options => EnableOption(options, GenerateOperatorsId),
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateEqualsAndGetHashCode)]
+        public async Task TestImplementIEquatable1()
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: Better names for tests would be nice (mention that the interesting part is that it's a struct for example)

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs [hunk 3]
```diff
@@ -1119,7 +1121,109 @@ public override bool Equals(object obj)
...
+               s == other.s;
+    }
+}",
+chosenSymbols: null,
+optionsCallback: options => EnableOption(options, ImplementIEquatableId),
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateEqualsAndGetHashCode)]
+        public async Task TestImplementIEquatable3()
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: What is this test trying to document?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: Good point on naming.  Will fix up.  To answer your question, in this specific case we are verifying that if you already implement IEquatable that we do not offer to implement that interface in the dialog.

**Pull request merged**